UnityPy/files/BundleFile.py

```diff
@@ -399,6 +396,89 @@ def save_fs(self, writer: EndianBinaryWriter, data_flag: int, block_info_flag: i
         writer.write_long(writer_end_pos)
         writer.Position = writer_end_pos
 
+
+    def save_web_raw(self, writer: EndianBinaryWriter):
+        # (version >= 4) hash
+        # (version >= 4) crc
+        # minimumStreamedBytes
+        # headerSize
+        # numberOfLevelsToDownloadBeforeStreaming
+        # levelCount
+        # compressedSize * levelCount
+        # uncompressedSize * levelCount
+        # (version >= 2) completeFileSize
+        # (version >= 3) file_info_header_size
+        # compressed assets
+        
+        if self.version > 3:
+            raise NotImplementedError("Saving Unity Web bundles with version > 3 is not supported")
+
+        # Calculate fileInfoHeaderSize for set offsets
+        file_info_header_size = 4  # for nodesCount
+        
+        for file_name in self.files.keys():
+            file_info_header_size += len(file_name.encode()) + 1  # +1 for null terminator
+            file_info_header_size += 4 * 2  # 4 bytes each for offset and size
+        
+        file_info_header_padding_size = 4 - (file_info_header_size % 4) if file_info_header_size % 4 != 0 else 0
+        file_info_header_size += file_info_header_padding_size
+
+        # Prepare directory info
+        directory_info_writer = EndianBinaryWriter()
+        directory_info_writer.write_int(len(self.files))  # nodesCount
+
+        file_content_writer = EndianBinaryWriter()
+        current_offset = file_info_header_size
+        
+        for file_name, f in self.files.items():
+            directory_info_writer.write_string_to_null(file_name)
+            directory_info_writer.write_u_int(current_offset)
+            
+            # Get file content
+            if isinstance(f, (EndianBinaryReader, EndianBinaryWriter)):
+                file_data = f.bytes
+            else:
+                file_data = f.save()
+            
+            file_size = len(file_data)
+            directory_info_writer.write_u_int(file_size)
+            
+            file_content_writer.write_bytes(file_data)
+            current_offset += file_size
+
+        directory_info_writer.write(b'\x00' * file_info_header_padding_size)
+        uncompressed_directory_info = directory_info_writer.bytes
+        uncompressed_file_content = file_content_writer.bytes
+
+        # Combine directory info and file content
+        uncompressed_content = uncompressed_directory_info + uncompressed_file_content
+        compressed_content = CompressionHelper.compress_lzma(uncompressed_content) if self.signature == "UnityWeb" else uncompressed_content
+        
+        # Write header
+        header_size = 60 # file_header 28, metadata max 4 * 8
```

Comment:
I don't think that it's good to assume a static value,
as it depends on the version as well as the length of version_player and version_engine that can be modified,
besides if I'm missing something.

So,
from the shared header parts:

string
+4
string
string

and for web itself:
version >= 4:
	+ 16
	+ 4

4
4
4
4 - node count -> nodes
	(4 * 2 * (n-1)) (skipped nodes)
	4
	4
version >= 2
	4
version >= 3
	4
------

version < 2
16 + 8 * n
version < 3
+4 = 20 + 8*n
version < 4
+4 = 24 + 8*n
version >= 4
+20 = 44 + 8*n

Reply:
Ah, I see, so you try to hide that via the padding, which is a good idea,
besides....if one of the earlier strings are too long due to modifications.